sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -494,8 +500,13 @@ case class DataSource(
             catalogTable = catalogTable,
             fileIndex = fileIndex)
         sparkSession.sessionState.executePlan(plan).toRdd
-        // Replace the schema with that of the DataFrame we just wrote out to avoid re-inferring it.
-        copy(userSpecifiedSchema = Some(data.schema.asNullable)).resolveRelation()
+        if (isForWriteOnly) {
+          // Exit earlier and return null
+          null
```

Comment:
I do not know whether returning null is ok here. This is based on a [similar early-exit solution](https://github.com/gatorsmile/spark/blob/5d38f09f47a767a342a0a8219c63efa2943b5d1f/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala#L178) used in `getOrInferFileFormatSchema`.